src/pybind/mgr/dashboard/services/access_control.py

```diff
@@ -24,6 +25,8 @@
 def password_hash(password, salt_password=None):
     if not password:
         return None
+    if six.PY2:
+        password = unicode(password, 'utf-8') if isinstance(password, str) else password
```

Comment:
What else can be `password` if not `str`? `unicode`... or `byte`?
```suggestion
    password = six.text_type(password)
```

Reply:
```ensure_text``` does the job as with ```text_type``` the test fails.

Reply:
In teuthology run we saw this failure: ```AttributeError: 'module' object has no attribute 'ensure_text'```.
It seems not all supported OS distributions have the minimum version ```six >= 1.12```.
So keeping the initial implementation.